Graphics/Slicer/Machine/Infill.hs

```diff
@@ -80,65 +80,68 @@ infillLineSegInside contour childContours line
+                  -- FIXME: we should 'stitch out' colinear segments, not just ignore them.
                   saneIntersection (Right PColinear)          (Left (HitStartPoint _ _    )) _                          = Nothing
                   saneIntersection _                          (Left (HitEndPoint   _ _    )) (Right PColinear)          = Nothing
                   saneIntersection _                          (Right PColinear)              _                          = Nothing
                   saneIntersection (Left NoIntersection)      (Left (HitEndPoint   _ point)) (Left NoIntersection)      = Just point
                   saneIntersection r1 r2 r3 = error $ "insane result of intersecting a line (" <> show myline <> ") with a contour " <> show c <> "\n" <> show r1 <> "\n" <> show r2 <> "\n" <> show r3 <> "\n"
-                  -- FIXME: we should 'stitch out' colinear segments, not just ignore them.
 
 -- Generate lines over entire print area, where each one is aligned with a -1 slope.
--- FIXME: other ways to only generate covering lines over the given contour?
 -- FIXME: assumes we're in positive space.
-coveringLineSegsNegative :: Contour -> ℝ -> [LineSeg]
-coveringLineSegsNegative (PointSequence contourPoints) ls = flip LineSeg s . f <$> [-xMin,-xMin+lsX..xMax]
-    where s = Point2 (xMaxOutside,yMaxOutside)
-          f v = Point2 (v,0)
+coveringLineSegsNegative :: Contour -> ℝ -> [PLine2]
+coveringLineSegsNegative (PointSequence contourPoints) ls = eToPLine2 . flip LineSeg slope . f <$> [0,lss..(xMax-xMinRaw)+(yMax-yMin)+lss]
+    where slope = Point2 (1,1)
+          f v = Point2 (v-xDiff,0)
+          xDiff = (-(xMin-yMax))
+          yMin = minimum $ yOf <$> contourPoints
+          yMax = maximum $ yOf <$> contourPoints
           xMinRaw = minimum $ xOf <$> contourPoints
-          xMin = head $ filter (> xMinRaw) [-ls, 0..]
+          xMin = head $ filter (> xMinRaw) [0, lss..]
           xMax = maximum $ xOf <$> contourPoints
-          yMax = maximum $ yOf <$> contourPoints
-          xMaxOutside = xMax + ls
-          yMaxOutside = yMax + ls
-          lsX = sqrt $ ls*ls+ls*ls
+          -- line spacing, taking into account the slope.
+          lss = sqrt $ ls*ls+ls*ls
 
 -- Generate lines over entire print area, where each one is aligned with a +1 slope.
--- FIXME: other ways to only generate covering lines over the given contour?
 -- FIXME: assumes we're in positive space.
-coveringLineSegsPositive :: Contour -> ℝ -> [LineSeg]
-coveringLineSegsPositive (PointSequence contourPoints) ls = flip LineSeg s . f <$> [0,lsY..yMax + xMax]
-    where s =  Point2 (xMaxOutside + yMaxOutside,- xMaxOutside - yMaxOutside)
-          f v = Point2 (0,v)
-          xMax = maximum $ xOf <$> contourPoints
+coveringLineSegsPositive :: Contour -> ℝ -> [PLine2]
+coveringLineSegsPositive (PointSequence contourPoints) ls = eToPLine2 . flip LineSeg slope . f <$> [0,lss..(xMax-xMin)+(yMax-yMin)+lss]
```

Comment:
I find the RHS hard to parse, it makes me need to think about the relative precedences of `.` and `<$>`. It would be much clearer if written with `fmap`.

Reply:
RHS?

Reply:
right hand side (of the equals sign)